test/e2e/clusterctl_upgrade.go

```diff
@@ -71,6 +71,8 @@ type ClusterctlUpgradeSpecInput struct {
 	// InitWithKubernetesVersion can be used to override the INIT_WITH_KUBERNETES_VERSION e2e config variable with a specific
 	// Kubernetes version to use to create the secondary management cluster, e.g. `v1.25.0`
 	InitWithKubernetesVersion string
+	// WorkloadKubernetesVersion is Kubernetes version used to create the workload cluster, e.g. `v1.25.0`
+	WorkloadKubernetesVersion string
```

Comment:
nit: let's move it down to the WorkloadFlavor

Reply:
My idea was (missed a detail when writing it up) to use this field if it is set and otherwise fallback to the KUBERNETES_VERSION env var (~ l.333)

Reply:
@sbueringer ah, I think you mean something like:

```
kubernetesVersion := ""
if input.WorkloadKubernetesVersion != "" {
	kubernetesVersion = input.WorkloadKubernetesVersion
} else {
	kubernetesVersion = input.E2EConfig.GetVariable(KubernetesVersion)
}
```

Reply:
To avoid confusion. In addition to that I meant to move the WorkloadKubenetesVersion down in the struct to the WorkloadFlavor

Reply:
Yeap, thanks and done.